codersdk/workspaces.go

```diff
@@ -26,27 +26,28 @@ const (
-	OwnerName                            string         `json:"owner_name"`
-	OwnerAvatarURL                       string         `json:"owner_avatar_url"`
-	OrganizationID                       uuid.UUID      `json:"organization_id" format:"uuid"`
-	OrganizationName                     string         `json:"organization_name"`
-	TemplateID                           uuid.UUID      `json:"template_id" format:"uuid"`
-	TemplateName                         string         `json:"template_name"`
-	TemplateDisplayName                  string         `json:"template_display_name"`
-	TemplateIcon                         string         `json:"template_icon"`
-	TemplateAllowUserCancelWorkspaceJobs bool           `json:"template_allow_user_cancel_workspace_jobs"`
-	TemplateActiveVersionID              uuid.UUID      `json:"template_active_version_id" format:"uuid"`
-	TemplateRequireActiveVersion         bool           `json:"template_require_active_version"`
-	LatestBuild                          WorkspaceBuild `json:"latest_build"`
-	Outdated                             bool           `json:"outdated"`
-	Name                                 string         `json:"name"`
-	AutostartSchedule                    *string        `json:"autostart_schedule,omitempty"`
-	TTLMillis                            *int64         `json:"ttl_ms,omitempty"`
-	LastUsedAt                           time.Time      `json:"last_used_at" format:"date-time"`
+	ID                                   uuid.UUID           `json:"id" format:"uuid"`
+	CreatedAt                            time.Time           `json:"created_at" format:"date-time"`
+	UpdatedAt                            time.Time           `json:"updated_at" format:"date-time"`
+	OwnerID                              uuid.UUID           `json:"owner_id" format:"uuid"`
+	OwnerName                            string              `json:"owner_name"`
+	OwnerAvatarURL                       string              `json:"owner_avatar_url"`
+	OrganizationID                       uuid.UUID           `json:"organization_id" format:"uuid"`
+	OrganizationName                     string              `json:"organization_name"`
+	TemplateID                           uuid.UUID           `json:"template_id" format:"uuid"`
+	TemplateName                         string              `json:"template_name"`
+	TemplateDisplayName                  string              `json:"template_display_name"`
+	TemplateIcon                         string              `json:"template_icon"`
+	TemplateAllowUserCancelWorkspaceJobs bool                `json:"template_allow_user_cancel_workspace_jobs"`
+	TemplateActiveVersionID              uuid.UUID           `json:"template_active_version_id" format:"uuid"`
+	TemplateRequireActiveVersion         bool                `json:"template_require_active_version"`
+	LatestBuild                          WorkspaceBuild      `json:"latest_build"`
+	LatestAppStatus                      *WorkspaceAppStatus `json:"latest_app_status"`
```

Comment:
Shouldn't this be `[]WorkspaceAppStatus`, as there can be multiple workspace apps each with a latest status? 

Alternatively, shouldn't this be a field on the agent itself?

Reply:
This is actually a field on the agent, but there's a specific reason this is here...

Ben reported and customers noted in our current aidev deployment that it's odd when you stop a workspace, you cannot see the last status that was reported. In the vast majority of cases, I think this is the best solution.

Simply always fetch the latest reported status and we can display it in the table view. That way even when stopped, we can show that the workspace created a PR or whatever.